pkg/build/api/validation/validation.go

```diff
@@ -367,32 +345,14 @@ func validateOutput(output *buildapi.BuildOutput) fielderrors.ValidationErrorLis
 func validateStrategy(strategy *buildapi.BuildStrategy) fielderrors.ValidationErrorList {
 	allErrs := fielderrors.ValidationErrorList{}
 
-	switch {
-	case len(strategy.Type) == 0:
-		allErrs = append(allErrs, fielderrors.NewFieldRequired("type"))
-
-	case strategy.Type == buildapi.SourceBuildStrategyType:
-		if strategy.SourceStrategy == nil {
-			allErrs = append(allErrs, fielderrors.NewFieldRequired("stiStrategy"))
-		} else {
-			allErrs = append(allErrs, validateSourceStrategy(strategy.SourceStrategy).Prefix("stiStrategy")...)
-		}
-
-	case strategy.Type == buildapi.DockerBuildStrategyType:
-		if strategy.DockerStrategy == nil {
-			allErrs = append(allErrs, fielderrors.NewFieldRequired("dockerStrategy"))
-		} else {
-			allErrs = append(allErrs, validateDockerStrategy(strategy.DockerStrategy).Prefix("dockerStrategy")...)
-		}
-
-	case strategy.Type == buildapi.CustomBuildStrategyType:
-		if strategy.CustomStrategy == nil {
-			allErrs = append(allErrs, fielderrors.NewFieldRequired("customStrategy"))
-		} else {
-			allErrs = append(allErrs, validateCustomStrategy(strategy.CustomStrategy).Prefix("customStrategy")...)
-		}
-	default:
-		allErrs = append(allErrs, fielderrors.NewFieldInvalid("type", strategy.Type, "type is not in the enumerated list"))
```

Comment:
Since the first thing we were doing was validating the strategy, we should move the call to `validateStrategy` up to where this code was.